sdk/core/core-util/src/aborterUtils.ts

```diff
@@ -61,3 +71,23 @@ export function createAbortablePromise<T>(
     abortSignal?.addEventListener("abort", onAbort);
   });
 }
+
+/**
+ * promise.race() implementation that aborts losers as soon as the first promise fulfills.
+ */
+export async function racePromisesAndAbortLosers<T>(
+  promises: ((abortOptions: AbortOptions) => Promise<T>)[],
+  abortSignal?: AbortSignalLike
+): Promise<T> {
+  const loserAborter = new AbortController();
+  const loserAbortListener = () => {
+    loserAborter.abort();
+  };
+  abortSignal?.addEventListener("abort", loserAbortListener);
+  const options = { abortSignal: loserAborter.signal, abortErrorMsg: "The operation was aborted." };
+
+  return Promise.race(promises.map((p) => p(options))).finally(() => {
+    loserAborter.abort();
+    abortSignal?.removeEventListener("abort", loserAbortListener);
```

Comment:
nitpick: should this be moved to inside the `loserAbortListener`/`abort` function instead?

Reply:
You need to specify the same type and same listener parameters to remove the added listener. So, no!
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener#matching_event_listeners_for_removal